File: src/main/java/facts/stmt/Assert.java
package facts.stmt;

import astnodes.exp.Expr;
import astnodes.pred.Pred;
import java.util.List;

public class Assert extends Stmt {
    private Pred pred;
    
    public Assert(String pc, Pred pred) {
        super(pc);
        this.pred = pred;
    }

    @Override
    public void replace(Expr oldExp, Expr newExp) {
        // TODO
    }

    @Override
    public List<Expr> getChildren() {
        // TODO
        return null;
    }
}

File: src/main/scala/astnodes/stmt/Assert.scala
package astnodes.stmt

import astnodes.exp.Expr
import astnodes.exp.variable.Variable
import astnodes.pred.Pred

case class Assert(override val pc: String, pred: Pred) extends Stmt(pc) {
  override def subst(v: Variable, w: Variable): Stmt = ???

  // TODO is there a way to merge these two methods
  override def toString: String = s"assert $pred;"
  override def toBoogieString: String = s"assert ${pred.toBoogieString};"
}
